src/sentry/static/sentry/app/views/settings/account/accountSecurity/components/recoveryCodes.jsx

```diff
@@ -0,0 +1,79 @@
+import {Box, Flex} from 'grid-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled, {css} from 'react-emotion';
+
+import {t} from '../../../../../locale';
+import Button from '../../../../../components/buttons/button';
+import Confirm from '../../../../../components/confirm';
+import EmptyMessage from '../../../components/emptyMessage';
+import InlineSvg from '../../../../../components/inlineSvg';
+import Panel from '../../../components/panel';
+import PanelBody from '../../../components/panelBody';
+import PanelHeader from '../../../components/panelHeader';
+import PanelItem from '../../../components/panelItem';
+
+const Code = styled(props => <PanelItem p={2} {...props} />)`
+  font-family: ${p => p.theme.text.familyMono};
+`;
+
+const PanelWarning = styled(props => <Flex p={2} {...props} />)`
+  ${p => css`
+    background-color: ${p.theme.alert.warning.backgroundLight};
+    border-bottom: 1px solid ${p.theme.borderLight};
+    color: ${p.theme.alert.warning.textDark};
+  `};
+`;
+
+class RecoveryCodes extends React.Component {
+  static propTypes = {
+    isEnrolled: PropTypes.bool,
+    codes: PropTypes.arrayOf(PropTypes.string),
+    onRegenerateBackupCodes: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    onRegenerateBackupCodes: () => {},
```

Comment:
Could we drop this? We use this component in one place and definitely provide it there